server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/SttpRequestBody.scala

```diff
@@ -26,17 +33,22 @@ class SttpRequestBody[F[_]](implicit ME: MonadError[F]) extends RequestBody[F, A
           case RawBodyType.InputStreamRangeBody => ME.unit(RawValue(InputStreamRange(() => new ByteArrayInputStream(bytes))))
           case _: RawBodyType.MultipartBody     => ME.error(new UnsupportedOperationException)
         }
-      case _ => throw new IllegalArgumentException("Stream body provided while endpoint accepts raw body type")
+      case Right(value) =>
+        bodyType match {
+          case mp: RawBodyType.MultipartBody =>
+            ME.unit(RawValue(extractMultipartParts(value.asInstanceOf[Seq[Part[client3.RequestBody[_]]]], mp)))
+          case _ => throw new IllegalArgumentException("Stream body provided while endpoint accepts raw body type")
+        }
     }
 
   override def toStream(serverRequest: ServerRequest, maxBytes: Option[Long]): streams.BinaryStream = body(serverRequest) match {
+    case Right(_: Seq[Part[client3.RequestBody[_]]]) => throw new IllegalArgumentException("Raw body provided while endpoint accepts stream body")
```

Comment:
this is a mess now: you call `body(req)` to get a `Right`, which you then further inspect and throw an exception if it doesn't fit. This needs cleaning up.